### PR TITLE
Added test framework including test for studio.js

### LINKED ISSUE #112: Set up test structure to fit in with edX's Jasmine testrunner
In order to add JavaScript tests, we first need to get the test infrastructure set up so that edX can run the tests

### LINKED ISSUE #114: Tests: Write tests for studio/cms
In particular, write tests for https://github.com/mitodl/edx-sga/blob/master/edx_sga/static/js/src/studio.js

We don't need 100% coverage, but there should be a test that exercises each function.

### PR DESCRIPTION
**Pull request by noisecapella**
Fixes #114 
Fixes #112

### CODE REVIEW

File: edx_sga/static/js/spec/main.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+(function(requirejs, define) {
+    requirejs.config({
+        paths: {
+            'jquery': 'lib/jquery-2.1.4.js',
+            'sinon': 'js/lib/sinon-1.16.1',
+            'URI': 'js/lib/URI',
+            'underscore': 'js/lib/underscore'
```
Comment (amir-qayyum-khan) [inline edx_sga/static/js/spec/main.js]: i am not sure but cant we access files like underscore from lms?

File: README.rst [hunk 2]
```diff
@@ -260,3 +260,10 @@ To get statement coverage::
 
     $ coverage run --source edx_sga manage.py lms --settings=test test edx_sga
     $ coverage report -m
+
+To run JavaScript tests, install js-test-tool_ and run::
+
+    $ js-test-tool run ./edx_sga/static/js_test.yml --use-firefox
```
Comment (amir-qayyum-khan) [inline README.rst]: Do we need to specify that user should be location below to run test (Devstack) ?
> /edx/app/edxapp/venvs/edxapp/src/edx-sga

Comment (noisecapella) [inline README.rst]: I'll get @carsongee's opinion on this

File: edx_sga/static/js/spec/main.js (hunk 1 above)
Comment (noisecapella) [inline edx_sga/static/js/spec/main.js]: I don't think we'll be able to access those javascript files, the infrastructure in edx-platform isn't in place to do that cleanly. We may do this in a future issue but for now we need to bundle everything we need here

File: README.rst (hunk 2 above)
Comment (noisecapella) [inline README.rst]: The tests can be run outside of the devstack currently so the relative path should work fine. I'll make an issue to get the test runner to work together with the test runner in the devstack.

**Pull request closed without merging**